controller/k8s/api.go

```diff
@@ -499,16 +471,11 @@ func GetNamespaceOf(obj runtime.Object) (string, error) {
 }
 
 // getNamespaces returns the namespace matching the specified name. If no name
-// is given, it returns all namespaces, unless the API was configured to only
-// work with a single namespace, in which case it returns that namespace. Note
-// that namespace reads are not cached.
+// is given, it returns all namespaces. Note that namespace reads are not
+// cached.
```

Comment:
Doesn't have to happen as part of this PR, but I originally removed caching for namespace reads when I implemented single-namespace, so in theory we could re-add caching now. Check out https://github.com/linkerd/linkerd2/commit/46c887ca0#diff-45847bbca5cd0ae2b05da10f9f5ab505 to see how it was previously setup.